wordpress-mobile/WordPress-iOS
# Integrate PHPickerViewController in Story Editor

Changes:

- Integrate `PHPickerViewController` in Story Editor
- No permission prompts are needed anymore. The only downside is that the "Add Photo" button will no longer be able to display a tiny preview of your latest photo from the gallery.
- Removes "Site Media" as a source (nobody using it; not worth re-implementing)
- Set standard maximum export resolution for images to 2048x2048 regardless of the device from which the story is created

There should be no other functional changes.
The goal is to decommission `WPMediaPicker` in this release.

To test:

🟡 This feature requires access to the camera and doesn’t seem to work in the simulator, even when selecting photos from the local library. I haven’t looked into it. Please test on the physical device.

To create a story, tap the "Create" button on the Home Screen and select "Story Post". Alternatively, add a "Story" block to an existing post.

- Verify that videos and images can be selected
- Verify that “Cancel” works
- Verify that the Editor no longer requires the permissions or asks for the permissions to access Photos (best tested using a clean build)
- Verify that adding photos works
- Verify that uploading both portrait and horizontal video works
- Verify that video longer than 5 minutes can’t be uploaded to free blogs
- Verify that errors are shown (hardcode in unc picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) 
- Verify that the story can be uploaded

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
